gbpservice/tests/contrib/nfp_scripts/upgrade_nfp.sh

```diff
@@ -0,0 +1,146 @@
+#! /bin/bash
+
+source upgrade_nfp.conf
+source $DEVSTACK_DIR/local.conf
+
+#TODO(DEEPAK): Should be retrieved from a result file populated by advanced mode.
+EXT_NET_NAME=ext-net
+
+function create_port_for_vm {
+    image_name=$1
+
+    GROUP="svc_management_ptg"
+    echo "GroupName: $GROUP"
+    PortId=$(gbp policy-target-create --policy-target-group $GROUP $InstanceName | grep port_id  | awk '{print $4}')
+
+    echo "Getting IpAddr for port: $PortId"
+    IpAddr_extractor=`neutron port-list|grep $PortId|awk '{print $11}'`
+    IpAddr_purge_last=${IpAddr_extractor::-1}
+    IpAddr=${IpAddr_purge_last//\"/}
+    echo "Collecting IpAddr : for $PortId"
+    echo $IpAddr 
+}
+
+function configure_vis_ip_addr_in_docker {
+    echo "Visibility VM IP address is: $IpAddr"
+    sed -i "s/VIS_VM_IP_ADDRESS/"$IpAddr"/" $NFPSERVICE_DIR/gbpservice/nfp/configurator/Dockerfile
+}
+
+function create_images {
+    source $DEVSTACK_DIR/openrc neutron service
+    unset OS_USER_DOMAIN_ID
+    unset OS_PROJECT_DOMAIN_ID
+
+    # prepare visibility image and upload it into glance
+    VISIBILITY_QCOW2_IMAGE=${VISIBILITY_QCOW2_IMAGE:-build}
+    VISIBILITY_QCOW2_IMAGE_NAME=visibility
+    InstanceName="VisibilityVM_instance"
+    create_port_for_vm $VISIBILITY_QCOW2_IMAGE_NAME
+
+    if [[ $VISIBILITY_QCOW2_IMAGE = build ]]; then
```

Comment:
This can either be built or be provided by the user.

Regards,
Deepak

On Thu, Jun 2, 2016 at 10:00 PM, njagadish <notifications@github.com> wrote:

> In gbpservice/tests/contrib/nfp_scripts/upgrade_nfp.sh
> <https://github.com/oneconvergence/group-based-policy/pull/414#discussion_r65573422>
> :
>
> > +    echo "Visibility VM IP address is: $IpAddr"
> > +    sed -i "s/VIS_VM_IP_ADDRESS/"$IpAddr"/" $NFPSERVICE_DIR/gbpservice/nfp/configurator/Dockerfile
> > +}
> > +
> > +function create_images {
> > +    source $DEVSTACK_DIR/openrc neutron service
> > +    unset OS_USER_DOMAIN_ID
> > +    unset OS_PROJECT_DOMAIN_ID
> > +
> > +    # prepare visibility image and upload it into glance
> > +    VISIBILITY_QCOW2_IMAGE=${VISIBILITY_QCOW2_IMAGE:-build}
> > +    VISIBILITY_QCOW2_IMAGE_NAME=visibility
> > +    InstanceName="VisibilityVM_instance"
> > +    create_port_for_vm $VISIBILITY_QCOW2_IMAGE_NAME
> > +
> > +    if [[ $VISIBILITY_QCOW2_IMAGE = build ]]; then
>
> Where is this VISIBILITY_QCOW2_IMAGE initialized? Is it always build?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/oneconvergence/group-based-policy/pull/414/files/2a7b140c1d5d64ce45d90ee2c20c960bd4a4bfd8#r65573422>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGq88xpee4yY0BqbBa8uDR9-53jYqETBks5qHwU4gaJpZM4IskzF>
> .
>